### PR TITLE
simplified points creation process

### PR DESCRIPTION
**Pull request by vlro**

### CODE REVIEW

File: tidepods/make_pfs.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    import clr
+
+    clr.AddReference('System')
+
+    import System
+
+    try:
+        sys.path.insert(0, sdkpath)
+
+    except System.IO.FileNotFoundException as exception:
```
Comment (dionhaefner) [inline tidepods/make_pfs.py]: Better just check whether the path exists with `os.path.isdir` or so beforehand

File: tidepods/make_pfs.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        msg = "Reference not found. Is the path to the sdk correct: '{0}'?".format(sdkpath)
+        raise ValueError(msg) from exception
+
+    clr.AddReference(r'DHI.PFS')
+
+    import DHI.PFS
+
+    sys.path.pop(0)
```
Comment (dionhaefner) [inline tidepods/make_pfs.py]: Can you wrap this in `try ... finally`?

```python
sys.path.insert(0, sdkpath)

try:
   ...

finally:
   sys.path.pop(0)
```

That way you ensure the `pop` even if an exception is thrown.

File: tidepods/make_pfs.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    Returns
+    -------
+    dfsfile :  str
+        Path to the generate dfs0 file within the tempdir
+
+    """
+
+    pfsfile = generatepfs(infile, date, mikepath, tempdir)
+    tidepredictor = os.path.join(mikepath, 'bin/x64/TidePredictor.exe')
+    cmd = tidepredictor + ' ' + pfsfile
```
Comment (dionhaefner) [inline tidepods/make_pfs.py]: You don't need to concatenate the command, just make it a list: `cmd = [tidepredictor, pfsfile]`. This has e.g. the advantage that you don't need to worry about quoting spaces.

File: tidepods/make_pfs.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    subprocess.check_call(cmd)
+
+    dfsfile = pfsfile.replace('.pfs', '.dfs0')
```
Comment (dionhaefner) [inline tidepods/make_pfs.py]: You should detect whether the command succeeded by checking if that file exists

File: tidepods/generate_pts.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    ----------
+    infile : str
+        File path to raster AOI.
+
+    Returns
+    -------
+    shp : shapely object
+
+    """
+    with rasterio.open(infile) as src:
```
Comment (dionhaefner) [inline tidepods/generate_pts.py]: For the record: You don't need `rasterio` for this (which is as heavyweight as Python packages get). You could achieve the same via `PyProj` or even `fiona`. But I think it's ok to keep that for now.

File: tidepods/create_tides.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    with fiona.open(outfile, 'w', crs=from_epsg(4326), driver='ESRI Shapefile',
+                    schema=pts_schema) as output:
+        for pid, (p, tv) in enumerate(zip(plist, tide_values)):
+            prop = {'p_ID': int(pid+1), 'tide_value': float(tv)}
+            output.write({'geometry': mapping(p), 'properties': prop})
+
+
+def main(infile, date, mikepath, outfile, **kwargs):
+    dirpath, filepath = os.path.split(infile)
+    tempdir = tempfile.mkdtemp(dir=dirpath)
```
Comment (dionhaefner) [inline tidepods/create_tides.py]: If you use the Python 3 construct

```python
with tempfile.TemporaryDirectory() as tempdir:
   ...
```

you don't need to clean up afterwards.

File: tidepods/create_tides.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+    except System.IO.FileNotFoundException as exception:
+        msg = "Reference not found. Is the path to the sdk correct: '{0}'?".format(sdkpath)
+        raise ValueError(msg) from exception
+
+    clr.AddReference(r'DHI.Generic.MikeZero.DFS')
+
+    import DHI.Generic.MikeZero.DFS
+
+    sys.path.pop(0)
```
Comment (dionhaefner) [inline tidepods/create_tides.py]: See below

File: tidepods/create_tides.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    import clr
+
+    clr.AddReference('System')
+
+    import System
+
+    try:
+        sys.path.insert(0, sdkpath)
+
+    except System.IO.FileNotFoundException as exception:
```
Comment (dionhaefner) [inline tidepods/create_tides.py]: See below

File: tidepods/create_tides.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    sdt = dfsfile.FileInfo.TimeAxis.StartDateTime
+    dfs_start_datetime = datetime.datetime(*(getattr(sdt, n) for n in ['Year', 'Month', 'Day',
+                                             'Hour', 'Minute', 'Second']))
+
+    diff = dfs_img_datetime - dfs_start_datetime
+    img_timestep = int(((diff.days * 24 * 60) + (diff.seconds / 60)) / timestep)
+
+    for i in range(dfsfile.ItemInfo.Count):
+        min_value = float(dfsfile.ItemInfo[i].MinValue)
+        acq_value = dfsfile.ReadItemTimeStep(i+1, img_timestep).Data[0]  # Value c.f. MSL
```
Comment (dionhaefner) [inline tidepods/create_tides.py]: Too little whitespace, should be `i + 1`. Please install `flake8` (or a `flake8` plugin for your editor of choice) to catch mistakes like this early on.

File: tidepods/cli.py [hunk 10]
```diff
@@ -36,50 +36,28 @@ def cli():
...
 
-    tidepods tides -p "C:/Program Files (x86)/DHI/2016/MIKE SDK/bin" -i C:/tides/aoi.dfs0
-    -d 20150131 -t 12:34 -s C:/tides/pts.shp -o C:/tides/pts_tides.shp
+    tidepods points -p "C:/Program Files (x86)/DHI/2016/MIKE SDK/bin" -i C:/tides/aoi.tif
+    -d 20150131 -t 10:30 -o C:/tides/pts_tides.shp
     """
     kwargs.update(date=dt.combine(date, timestamp))
-    from . import write_tides as wt
-    wt.write_tide_values(**kwargs)
+    from . import create_tides as ct
```
Comment (dionhaefner) [inline tidepods/cli.py]: We don't use the `from . import` notation if we can avoid it. Absolute imports are better: `from tidepods import`

File: setup.py [hunk 11]
```diff
@@ -2,9 +2,9 @@
 
 setup(
     name='tidepods',
-    version='0.1',
+    version='1.0',
```
Comment (dionhaefner) [inline setup.py]: ... I wouldn't call it `1.0` until it's tried and tested extensively, but you're the maintainer :wink:

Comment (j08lue) [reply setup.py]: Also, I think we use MAJOR.MINOR.PATCH (3 places) now everywhere, don't we? https://semver.org/

Major versions are for breaking changes. So if you call something 1.0, you indicate that you are confident that your API (the major function signatures and choice of abstractions) is well thought through and will be stable for a while - which you only know after extensive trial, like @dionhaefner said. :-)

File: tidepods/make_pfs.py (hunk 1 above)
Comment (vlro) [reply tidepods/make_pfs.py]: Ok done

File: tidepods/make_pfs.py (hunk 2 above)
Comment (vlro) [reply tidepods/make_pfs.py]: Done

File: tidepods/make_pfs.py (hunk 3 above)
Comment (vlro) [reply tidepods/make_pfs.py]: Noted

File: tidepods/make_pfs.py (hunk 4 above)
Comment (vlro) [reply tidepods/make_pfs.py]: This is handled once we attempt to read the dfs file in `create_tides.py`,

```python
    dfsfilepath = make_pfs.make_dfs0(infile, date, mikepath, tempdir)
    dfsfile = DHI.Generic.MikeZero.DFS.DfsFileFactory.DfsGenericOpen(dfsfilepath)

```

 and it throws up an error if the file doesn't exist. Should we explicitly check here as well?

File: tidepods/generate_pts.py (hunk 5 above)
Comment (vlro) [reply tidepods/generate_pts.py]: Kept for now. Moved the rasterio import to within the function so it is only imported in cases where the input file is a raster. Else fiona is imported if the input file is a vector.

File: tidepods/create_tides.py (hunk 7 above)
Comment (vlro) [reply tidepods/create_tides.py]: Done

File: tidepods/create_tides.py (hunk 9 above)
Comment (vlro) [reply tidepods/create_tides.py]: Done, but according to [this](https://stackoverflow.com/a/37956466) the rules was changed in 2012

File: setup.py (hunk 11 above)
Comment (vlro) [reply setup.py]: Noted, updated to MAJOR.MINOR.PATCH and back to 0.1.0 for now

File: tidepods/cli.py (hunk 10 above)
Comment (vlro) [reply tidepods/cli.py]: Sorted

File: tidepods/make_pfs.py (hunk 4 above)
Comment (dionhaefner) [reply tidepods/make_pfs.py]: Yes. The function does not know how the result is used afterwards, so a check belongs here. If the function succeeds, you should be able to assume that everything went well.

File: tidepods/create_tides.py (hunk 9 above)
Comment (dionhaefner) [reply tidepods/create_tides.py]: Interesting, I didn’t know that.

**Pull request merged**